utility/subcommands.py

```diff
@@ -0,0 +1,264 @@
+- run(log_directory, subcomponent_filter, output_directory): Orchestrates log file processing and output storage.
+
+### **Folder Structure:**
+The script organizes extracted outputs into the specified output directory as follows:
+
+"""
+
+import hashlib
+import json
+import os
+import re
+
+from docopt import docopt
+
+DOC = """
+Standard script to fetch and process log files from a given directory.
+
+Usage:
+    subcommands.py --logdir <log_directory> --filter <subcomponent_filter> --outdir <output_directory>
+    subcommands.py (-h | --help)
+
+Options:
+    -h --help                      Show this help message
+    --logdir <log_directory>       Directory containing log files
+    --filter <subcomponent_filter> Filter logs by subcomponent (e.g., rgw, rbd, rados)
+    --outdir <output_directory>    Directory where output JSON files will be stored
+"""
+
+
+def compute_output_hash(output):
+    """
+    Computes a SHA-256 hash for the given output data.
+    Args:
+        output (dict): The output data to hash.
+    Returns:
+        str: The computed hash value as a hexadecimal string.
+    """
+    return hashlib.sha256(
+        json.dumps(output, sort_keys=True).encode("utf-8")
+    ).hexdigest()
+
+
+def clean_log_line(line):
+    """
+    Cleans log lines by removing timestamps and log level prefixes.
+    Args:
+        line (str): A single line from the log file.
+    Returns:
+        str or None: The cleaned log line or None if empty.
+    """
+    line = re.sub(
+        r"^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2},\d{3} - cephci - ceph:\d+ - (INFO|DEBUG|ERROR) -\s*",
+        "",
+        line,
+    ).strip()
+    line = re.sub(
+        r"^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2},\d{3} (INFO|DEBUG|ERROR):?\s*", "", line
+    ).strip()
+    return line if line else None
+
+
+def reconstruct_json(lines):
+    """
+    Reconstructs JSON data from cleaned log lines.
+    Args:
+        lines (list): A list of cleaned log lines.
+    Returns:
+        dict or list: The reconstructed JSON data if valid, otherwise a list of cleaned lines.
+    """
+    cleaned_lines = [clean_log_line(line) for line in lines if clean_log_line(line)]
+    try:
+        return json.loads("\n".join(cleaned_lines))
+    except json.JSONDecodeError:
+        return cleaned_lines
+
+
+def extract_radosgw_admin_commands(log_lines):
+    """
+    Extracts `radosgw-admin` commands and their corresponding outputs from log files.
+    Args:
+        log_lines (list): A list of log file lines.
+    Returns:
+        dict: Extracted commands and their outputs, preventing duplicates using SHA-256 hashes.
+    """
+    results, existing_hashes = [], set()
+    current_command, current_output_lines = None, []
+
+    for line in log_lines:
+        cleaned_line = clean_log_line(line)
+        if not cleaned_line:
+            continue
+
+        cmd_match = re.search(r"(?:sudo )?radosgw-admin[^;\n]+", cleaned_line)
```

Comment:
Can we redesign... this will cause issues as the Jenkins agents don't have large RAM memory.